tests/stack/test_api.py

```diff
@@ -529,3 +572,83 @@ def test_download_resource(self, stack_api):
+            when valid stack configurations and stack status are fed in to stack_api.deploy_config.
+        """
+        test_deploy_output = {'test': 'test'}
+        # Setup mocks for job resource deployment
+        stack_api._update_job = mock.MagicMock()
+        stack_api._update_job.return_value = 12345
+        stack_api._put_job = mock.MagicMock()
+        stack_api._put_job.return_value = 12345
+        stack_api.jobs_client.get_job = mock.MagicMock()
+        stack_api.jobs_client.get_job.return_value = test_deploy_output
+
+        # Setup mocks for workspace resource deployment
+        stack_api.workspace_client.import_workspace = mock.MagicMock()
+        stack_api.workspace_client.import_workspace_dir = mock.MagicMock()
+        stack_api.workspace_client.client.get_status = mock.MagicMock()
+        stack_api.workspace_client.client.get_status.return_value = test_deploy_output
+
+        # Setup mocks for dbfs resource deployment
+        stack_api.dbfs_client.cp = mock.MagicMock()
+        stack_api.dbfs_client.client = mock.MagicMock()
+        stack_api.dbfs_client.client.get_status.return_value = test_deploy_output
+
+        # Create files and directories associated with workspace and dbfs resources to ensure
+        # that validations within resource-specific services pass.
+        test_stack = copy.deepcopy(TEST_STACK)
+        for resource in test_stack[api.STACK_RESOURCES]:
+            resource_service = resource[api.RESOURCE_SERVICE]
+            resource_properties = resource[api.RESOURCE_PROPERTIES]
+            curr_source_path = resource_properties.get(api.DBFS_RESOURCE_SOURCE_PATH, '')
+            resource_properties.update(
+                {api.DBFS_RESOURCE_SOURCE_PATH: os.path.join(tmpdir.strpath, curr_source_path)})
+            if resource_service == api.WORKSPACE_SERVICE:
+                if workspace_api.NOTEBOOK == \
+                        resource_properties[api.WORKSPACE_RESOURCE_OBJECT_TYPE]:
+                    os.makedirs(os.path.dirname(resource_properties[
+                                                api.WORKSPACE_RESOURCE_SOURCE_PATH]))
+                    with open(resource_properties[api.WORKSPACE_RESOURCE_SOURCE_PATH], 'w') as f:
+                        f.write("print('test')\n")
+                if resource_properties[api.WORKSPACE_RESOURCE_OBJECT_TYPE] == \
+                        workspace_api.DIRECTORY:
+                    os.makedirs(resource_properties[api.WORKSPACE_RESOURCE_SOURCE_PATH])
+            elif resource_service == api.DBFS_SERVICE:
+                if resource_properties[api.DBFS_RESOURCE_IS_DIR]:
+                    os.makedirs(resource_properties[api.DBFS_RESOURCE_SOURCE_PATH])
+                else:
+                    with open(resource_properties[api.DBFS_RESOURCE_SOURCE_PATH], 'w') as f:
+                        f.write("print('test')\n")
+
+        # Run deploy command on stack and validate stack status. If there are no exceptions
```

Comment:
It seems now you also check the content of the status right? So it is not just validate schemas

Reply:
Yeah I'll clarify that in the comments.